File: hogworz/HomeWork/src/main/java/pages/MainPage.java
package pages;

import driver.Driver;
import org.openqa.selenium.By;

import java.net.MalformedURLException;
import java.util.concurrent.TimeUnit;

public class MainPage extends BasePage {

    By SeacherBar = useLoc("tv_search");
    //自选
    By ZiXuan = useLoc("//*[@text='自选']");

    /**
     * 启动driver ,到主页面
     * @return
     */
    public static MainPage goMainPage(){
        try{
            Driver.start();
            Driver.getdriver().manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
        }catch (MalformedURLException e){
            e.printStackTrace();
        }
        return new MainPage();
    }

    /**
     * 从主页点击搜索框到搜索页面
     */
    public SearchPage goSearchPage(){
        find(SeacherBar).click();
        return new SearchPage();
    }

    /**
     * 到自选页面
     * @return
     */
    public ZiXuanPage goZiXuanPage(){
        find(ZiXuan).click();
        return new ZiXuanPage();
    }


}

File: PycharmProjects/xueqiuAppium/pages/mainpage.py
#-*- coding:utf-8 -*-
from pages.basepage import BasePage
from driver.appium_driver import Driver
from appium.webdriver.common.mobileby import By
from pages.searchpage import SearchPage


class MainPage(BasePage):
    #搜索框
    search_loc = (By.ID,"tv_search")
    @staticmethod
    def go_main_page():
        Driver.init()
        return MainPage()
    
    def go_search_page(self):
        self.find(*self.search_loc).click()
        return SearchPage()